base/generator.jl

```diff
@@ -52,6 +52,9 @@ size(g::Generator) = size(g.iter)
 axes(g::Generator) = axes(g.iter)
 ndims(g::Generator) = ndims(g.iter)
 
+getindex(g::Generator, I...) = map(g.f, g.iter[I...])
```

Comment:
IIUC, this is broken for something like `(tuple(x) for x in ["a"])[1]` and `(tuple(x) for x in [[0]])[1]`. I think it's better to avoid defining "vectorized" `getindex` for `Generator` (and let something like #30845 handle it).

Reply:
Thanks for pointing this out. Using broacasting would fix your first example but not the second. I think we'll need to address this in some way as the differences can occur with linear indexing as shown in your examples. Ideally we want `g[i...] == collect(g)[i...]`.

Reply:
I've revised the code to take into account the index shape